src/components/structures/MatrixChat.js

```diff
@@ -1930,4 +2058,220 @@ export default createReactClass({
             {view}
         </ErrorBoundary>;
     },
+
+    async addLiveEventToIndex(ev) {
```

Comment:
I wonder if it would make sense to move most of this code that's been added to `MatrixChat` into a separate file/class, given how large it is already, and how independent the seshat integration seems.

Reply:
Yes, that should happen. It's kept mostly in a single file for this PR to make it easier to review. @dbkr mentioned that already and I agreed.